dev/io.openliberty.microprofile.reactive.messaging.internal_fat/fat/src/io/openliberty/microprofile/reactive/messaging/fat/apps/emitter/EmitterRestResource.java

```diff
@@ -0,0 +1,61 @@
+/*******************************************************************************
+ * Copyright (c) 2023 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ ******************************************************************************/
+
+package io.openliberty.microprofile.reactive.messaging.fat.apps.emitter;
+
+import jakarta.inject.Inject;
+import jakarta.ws.rs.POST;
+
+import jakarta.ws.rs.Path;
+import org.eclipse.microprofile.reactive.messaging.Channel;
+import org.eclipse.microprofile.reactive.messaging.Emitter;
+import org.eclipse.microprofile.reactive.messaging.Message;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+@Path("/")
+public class EmitterRestResource<T> {
+
+    public static final String PAYLOAD_CHANNEL_NAME = "restful-emitter-payload";
+    public static final String MESSAGE_CHANNEL_NAME = "restful-emitter-message";
+
+    @Inject
+    @Channel(PAYLOAD_CHANNEL_NAME)
+    Emitter<String> payloadEmitter;
+
+    @Inject
+    @Channel(MESSAGE_CHANNEL_NAME)
+    Emitter<String> messageEmitter;
```

Comment:
Use `private` on injected fields unless you're deliberately allowing them to be accessed from elsewhere.